Add tests for Celo block receipt

alecps opened the pull request:
Had to make some changes to the API handlers to make these tests work. These changes actually didn't exist in celo-blockchain which indicates these api methods have never supported fetching the celo block receipt to begin with. Not sure if we'll want to keep all of this but it works!

Commit:
adds tests and changes to support fetching celo block receipt

piersy committed:
Remove second test (#82)

Rather than having 2 tests one that exercises eth_getBlockReceipts and another that exercises eth_getTransactionReceipt this change incorporates both rpc calls into the same test and checks that they produce matching output.

piersy reviewed (approved):
Thank you @alecps this looks good, yep it seems that the block receipts RPC is broken for celo I created an issue for that [here](https://github.com/celo-org/celo-blockchain/issues/2293). I also merged a small PR that just combined the 2 tests so that we exercise all RPCs tested so far in a single test.

piersy commented on internal/ethapi/api.go:
```diff
@@ -1989,6 +1993,24 @@ func (s *TransactionAPI) GetRawTransactionByHash(ctx context.Context, hash commo
 func (s *TransactionAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index, err := s.b.GetTransaction(ctx, hash)
 	if tx == nil || err != nil {
+		block, err := s.b.BlockByHash(ctx, hash)
```
In celo we added an extra rpc (`eth_getBlockReceipt`) I think we should maintain this approach as opposed to overloading `eth_getTransactionReceipt`.

piersy committed:
Add check to ensure we actually got a block

At some point the datadir I was working against got corrupted and I
couldn't retrieve any blocks, however the test was still passing.

piersy committed:
Add eth_getBlockReceipt rpc method

Update the test to use this to get the block receipt.
Remove mechanism to get block receipt through transaction receipt rpc.
Update marshalBlockReceipt to match the output on celo-blockchain.
